PORTAL/jobs/__main__.py

```diff
@@ -1143,11 +1152,22 @@ def main(
             reqid = parsed
         cmd_kwargs['reqid'] = reqid
     elif cmd.startswith('request-'):
-        reqid = RequestID.generate(cfg, user, kind=cmd[8:])
+        reqid = RequestID.generate(
+            cfg,
+            user,
+            kind=cmd[8:],
+            workerid=cmd_kwargs["worker"]
+        )
         cmd_kwargs['reqid'] = reqid
     else:
         reqid = None
 
+    if _should_ensure_next(cmd):
+        if reqid is not None:
+            jobs.ensure_next(reqid.workerid)
+        else:
+            jobs.ensure_next(cmd_kwargs["workerid"])
```

Comment:
I'm not clear on what's happening here.  The point of `jobs.ensure_next()` is to make sure the queue is getting cleared out, in case the last job didn't start the next one.  With multiple queues, it would need to do that for all the queues.  It shouldn't apply to any one queue.

Reply:
I guess I'm still confused by this -- if the user submits a job on the `mac` queue, it seems like you'd want to clear that one, and leave the other one alone, no?

Reply:
`jobs.ensure_next()` is a non-blocking safety net in case a finishing job fails to kick off the next one (from its queue).  The more often we call it, the more likely we will avoid idling workers while there is work to be done (regardless of queue).  If we  only check the worker that the current command is targeting then we miss out on our chance to fix any other hung queue.

Reply:
Cool, that's helpful.  I had thought it was "do the work the current task depends on first", but it sounds like that's not it.  Easy enough to loop over the queues and call this on all of them.

Reply:
> Easy enough to loop over the queues and call this on all of them.

exactly

Reply:
It seemed more correct to do this iteration inside `Jobs.ensure_next` instead of here...

Reply:
agreed